README.md

```diff
@@ -1 +1,62 @@
-# Readme
+# VS Verification Toolbox
+
+This module provides several useful tools for writing VS Code extensions that verify code.
+
+## Features
+
+### General Utilities
+
+The `util` package offers several convenient utilities and wrappers.
+
+#### File System Locations
+
+A `Location` represents a directory on the file system, providing a simple API for navigating through the hierarchy. Example usage:
+
+```typescript
+const root = new Location("/tmp/folder");
+root.basePath; // "/tmp/folder"
+root.enclosingFolder; // Location(/tmp)
+root.child("sub", "path"); // Location(/tmp/folder/sub/path)
+root.path("sub", "path"); // "/tmp/folder/sub/path"
+root.executable("run"); // unix-based: "/tmp/folder/run"; windows: "/tmp/folder/run.exe"
+await root.mkdir(); // ensures the folder exists
+await root.exists(); // true
+```
+
+#### Platform Checking
+
+`Platform` is a simple enum representing the three major platforms: `Linux`, `Windows`, `Mac`.
+
+It also provides a constant `currentPlatform` equal to the current platform, or `null` if it's not one of those three.
+
+#### Progress Reporting
+
+The `ProgressListener` type describes a callback for reporting progress from long-running asynchronous tasks. It takes the fraction of the task that is complete, as well as a user-facing description of the current step.
+
+Additionally, there is a very convenient function `withProgressInWindow` that runs an asynchronous task, forwarding the progress it reports (to the given `ProgressListener`) to the VS Code window as a notification with a nice progress bar.
+
+### Dependency Installation
+
+The `dependencies` package provides a way to define your dependencies declaratively and install them asynchronously.
+
+An example dependency might be defined as follows:
+
+```typescript
+const myDependency = new Dependency<"remote" | "local">(
+    "/path/to/install/folder",
+    ["remote",
+        new InstallerSequence([
+            new FileDownloader("https://remote.com/file.zip"),
+            new ZipExtractor("unzipped"),
+        ])
```

Comment:
`FileDownloader + ZipExtractor` is a common combination. What about providing a `RemoteZipSource` or `RemoteZipDownloader` that internally uses the `InstallerSequence`?

Reply:
That's a good idea. It would definitely require `InstallerSequence` to flatten nested sequences somehow, though—I wonder how that could best be achieved…
An easy solution might be to not make `RemoteZipSource`/`RemoteZipDownloader` actual classes, instead just having them be methods that return an `InstallerSequence`, since that would be easy enough to flatten.